lib/dialects/abstract/query-generator.js

```diff
@@ -1520,6 +1522,195 @@ module.exports = (function() {
       return result;
     },
 
+    whereQuery: function(where, options) {
+      var query = this.whereItemsQuery(where, options);
+      if (query && query.length) {
+        return 'WHERE '+query;
+      }
+      return '';
+    },
+    whereItemsQuery: function(where, options, binding) {
+      if (
+        (Array.isArray(where) && where.length === 0) ||
+        (_.isPlainObject(where) && _.isEmpty(where)) ||
+        where === '' ||
+        where === null ||
+        where === undefined
+      ) {
```

Comment:
It probably could. For some reason i prefer the verbosity here to exactly communicaet what cases are no-op.